libraries/integration/Microsoft.Bot.Builder.ApplicationInsights.Core/TelemetryBotIdInitializer.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.ApplicationInsights.Core
+{
+    /// <summary>
+    /// Initializer that sets the user ID based on Bot data.
+    /// </summary>
+    public class TelemetryBotIdInitializer : ITelemetryInitializer
+    {
+        private readonly IMemoryCache _memoryCache;
```

Comment:
⚠️ If you were to keep this implementation (which I don't suggest you do)... ⚠️ 

Remove this, you'll now just pull the value out of the `HttpContext::Items` collection.

Add the field that will be the key to the collection:

```CSharp
        internal static readonly object HttpContextItemsKey = "__MSBOTFRAMEWORK_TELEMTRYBOTIDINITIALIZER_BODY_JSON_STRING";
```